src/afrolabs/components/kafka.clj

```diff
@@ -1889,7 +1912,24 @@ Supports a timeout operation."))
                   (deliver has-caught-up-once true)
                   (csp/close! caught-up-ch))
 
-        topic-partition-offset-ch (csp/chan)
+        ;; topic-partition-offset-ch (csp/chan) ;; why was this here?
+
+        ;; `caught-up-once?` mechanism is a way to block the initial ktable (deref) call until the ktable consumer
+        ;; was current at least once, relative to when the consumer started.
+        ;; latest-offsets are checked initially, and when the consumer's current-offset surpasses this, the
+        ;; check is passed. It is special because it does not check if the consumer is fully
+        ;; current all the time, but instead checks if the consumer caught up to where the offsets were
+        ;; at the time of starting. It is a type of "caught up enough to be useful" check.
+
+        ;; If `caught-up-once?` is false, then the first deref will only return if the consumer's
+        ;; current-offset is the same as latest-offset, which could be tricky of a producer is busy producing
+        ;; records to this topic while the first deref is waiting.
+
+        fully-current-notification-ch (csp/chan (csp/sliding-buffer 1))
+        caught-up-notifications-strategy (apply CaughtUpNotifications
+                                                (remove nil? [fully-current-notification-ch
+                                                              (when-not caught-up-once?
+                                                                caught-up-ch)]))
```

Comment:
For ktables, we will now always add the `CaughtUpNotifications` strategy, because we will use it to wait during the `ktable-wait-until-fully-current` implementation. It is also _sometimes_ used during system startup, to wait until a ktable is fully loaded, before continuing, typically during component init.